### DISCUSSION TITLE
Convergence of electrical circuit depend on ground position.

### DISCUSSION THREAD

**Discussion by georgtree** [OpenModelica/OpenModelica#11676] [Q&A]
Hello!
I faced with strange problem, I have this circuit (can't publish sources here for MOSFET unfortunately):
![image](https://github.com/OpenModelica/OpenModelica/assets/113975847/ca0c91b9-ca52-4364-bdb4-56fbae550c0f)

In such configuration it works and produces meaningful results:
![image](https://github.com/OpenModelica/OpenModelica/assets/113975847/60f3d4d8-af8c-4318-b3c2-8485e4424117)
![image](https://github.com/OpenModelica/OpenModelica/assets/113975847/47b588df-9275-4fbf-9b17-eb73609ca96a)

But if I slightly change configuration to this (move ground connection), it fails to converge:
![image](https://github.com/OpenModelica/OpenModelica/assets/113975847/7b195c07-3a98-44a4-8cea-0c1584e17311)
with this error:

```
Failed to solve linear system of equations (no. 96) at time 0.000000, system is singular for U[2, 2].
The default linear solver fails, the fallback solver with total pivoting is started at time 0.000000. That might raise performance issues, for more information use -lv LOG_LS.
Total Pivot: Matrix (nearly) singular at initialization.
```

I understand that there is something with process of building matrix for linear solver, but in which direction I should go to find an issue?
Ideally, solver should detect singularity and try to rearrange matrix to eliminate this (in SPICE it usually adds GMIN conductance to all nodes)
Thank you in advance.

**Comment by casella** [comment]
@georgtree please use version 1.22.0 or, even better, the forthcoming 1.22.1 release. The connection management is much improved there. In all likelyhood there is some bogus connection in your systems that will be revealed.

Please let me know if that worked. Thanks!

**Comment by georgtree** [comment]
Hello!
I installed latest release version, but the issue still persist.
If I add a small capacitor in parallel to inductance, all works as it should:
![image](https://github.com/OpenModelica/OpenModelica/assets/113975847/b38eeea7-e228-4223-833c-3fcdab284690)

So, what causes the issue, what do you think?

**Comment by casella** [comment]
OK, this is strange, because the equations should be equivalent no matter what the position of the ground is.

Maybe it's a problem with tearing, can you add `--tearingMethod=minimalTearing` to Simulation Setup | Translation Flags | Additional Translation flags in both cases and report?
